src/figuresClasses.ts

```diff
@@ -1,19 +1,78 @@
-export interface Figure {
+type Colors = 'red' | 'green' | 'blue';
 
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ([a, b, c].some((s) => s <= 0)
+    || a + b <= c
+    || a + c <= b
+    || b + c <= a
+    ) {
+      throw new Error('Invalid triangle');
```

Comment:
it will be better to give more meaningful error messages to let user understand what's wrong